### PR TITLE
fixed pin defines, added details

### PR DESCRIPTION
**Pull request by maclobdell**
added details of how to find out what how the definitions for pins are set in target.json when using YOTTA_CFG defines in code.

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -21,9 +21,16 @@ To build and run this example the following requirements are necessary:
...
 
-2. To know where on your board those pins are, please look at its [pinout picture](http://developer.mbed.org/platforms/FRDM-K64F/#overview).
+2. These pin definitions are set in the yotta target's ``target.json`` file which can be found at ``yotta_targets/frdm-k64f-gcc/target.json`` in this example.  Each keyword in the define represents a level in the target descriptor structure.  Under config...hardware...test-pins...serial, the pin port numbers are defined.
+```
+"serial": {
+"tx": "PTC17",
+"rx": "PTD2"
+}
+```
+3. To know where on your board those pins are, please look at its [pinout picture](http://developer.mbed.org/platforms/FRDM-K64F/#overview).
```
Comment (0xc0170) [inline Readme.md]: Can you fix the formatting of the code (indentation) and then 3rd is not displayed properly.

**Comment by 0xc0170** [comment]
Thanks, I was planning to add this to readme ,all asynch examples needs this update to latest test-pins config. You might be faster than me.

File: Readme.md [hunk 2]
```diff
@@ -21,9 +21,16 @@ To build and run this example the following requirements are necessary:
 
 TX and RX pin definitions are target specific - the code and the physical board must both be using the same pins:
 
-1. To know which pins the code is looking for, see the ``serial_asynch.cpp`` file for this sample. The pins are defined as ``TEST_SERIAL_ONE_TX_PIN`` and ``TEST_SERIAL_TWO_RX_PIN``.
+1. To know which pins the code is looking for, see the ``serial_asynch.cpp`` file for this sample. The pins are defined as ``YOTTA_CFG_HARDWARE_TEST_PINS_SERIAL_TX`` and ``YOTTA_CFG_HARDWARE_TEST_PINS_SERIAL_RX``.
```
Comment (0xc0170) [inline Readme.md]: We could removed 1. , as pins were defined in the code file, they are now in the config, which is already explained in the 2nd. Shall we merge 1 and 2?

**Comment by maclobdell** [comment]
I revised it again.  Please let me know what you think.  This look better?

**Comment by maclobdell** [comment]
looks like the 3rd item and indenting is still messed up.

**Comment by maclobdell** [comment]
1. First, find the pin references in the source files.  In this example, view ``serial_asynch.cpp``. The pin references are ``YOTTA_CFG_HARDWARE_TEST_PINS_SERIAL_TX`` and ``YOTTA_CFG_HARDWARE_TEST_PINS_SERIAL_RX``.

2. Next, find the target-specific pin definitions corresponding to the pin references in the target's ``target.json`` file.   In this example, the file is located at ``yotta_targets/frdm-k64f-gcc/target.json``.

 Each keyword in the pin reference represents a level in the target configuration structure.  To find what YOTTA_CFG_HARDWARE_TEST_PINS_SERIAL resolves to, look under config...hardware...test-pins...serial.

 In this example, the serial tx and rx pins are defined as:
 ```
 "serial": {
 "tx": "PTC17",
 "rx": "PTD2"
 }
 ```
3. Finally, locate the pins (i.e. PTC17, PTD2) on the board's [pinout

**Comment by 0xc0170** [comment]
Shall we merge 1st and 2nd?

can you indent the code snippet from json file, same as it is in config ?

```
"serial": {
    "tx": "PTC17",
    "rx": "PTD2"
}
```

**Comment by maclobdell** [comment]
okay, I fixed the indentation and amended my commit and pushed it back to my fork.  Is it okay?  do you want to merge in my pull request or tweak more first?

**Comment by maclobdell** [comment]
oh, by the way, I think it is good to keep 1st and 2nd steps separate, because they are really two different things you have to do.  first open one file to see what thing you need to look for, then open the second file to find that thing.  just my thought.  you can override.  :)

**Comment by 0xc0170** [comment]
LGTM

we need to update other examples as well

**Pull request merged**